content/en/docs/performance/resource_consumption.md

```diff
@@ -0,0 +1,188 @@
+---
+draft: false
+linktitle: Kmesh Resource Consumption
+menu:
+  docs:
+    parent: performance
+    weight: 2
+title: Kmesh Resource Consumption
+toc: true
+type: docs
+---
+# 1. Background Information
+eBPF programs consume a certain amount of CPU and memory resources during their execution, which may affect the performance of other applications. To gain a deeper understanding of Kmesh’s resource consumption under different loads, we conducted several CPU and memory stress tests. These tests aim to evaluate the limits of CPU and memory consumption of Kmesh in real-world usage scenarios.
+
+**This documentation is based on Kmesh 0.4 ads mode**
+# 2. Environment setup
+![resource_env](/docs/performance/resource_test_env.png)
+
+- K8S version: v1.27
+- Kmesh version: 0.4 ads mode
+- Kernel version: 5.10
+- Node flavor: 8U16G
+- Testing tool: fortio
+- Metric collecting tools: [bpftop](https://github.com/Netflix/bpftop), [inspektor-gadget](https://github.com/inspektor-gadget/inspektor-gadget)
+
+
+# 3. Test case 1 - POD with CPU limit scenario，collect Kmesh eBPF CPU usage
+## 3.1 Deploy 1 App A and set App A POD's CPU limit，then generage load and collect Kmesh eBPF CPU usage
+![resource_test1](/docs/performance/resource_test1.png)
+
+Set the CPU limit for App A to 1 (1 CPU), and collect corresponding Kmesh eBPF CPU consumption.
+
+**Note: With the system having 8 cores and a CPU limit of 1, it means the POD can consume up to 12.5% of the CPU.**
+
+Generate load using the fortio client.
+```sh
+#!/bin/bash
+
+client_address=`kubectl get pod | grep fortio-client | awk {'print $1'}`
+echo "$client_address" | xargs -P 0 -I {} kubectl exec -it {} -- fortio load -quiet -c 1500 -t 100s -qps 0 -keepalive=false fortio-server.default.svc.cluster.local:80
+```
+
+Collect CPU usage using [bpftop](https://github.com/Netflix/bpftop)
+```sh
+$ ./bpftop
+```
+
+Testing result：
+![resource_result1](/docs/performance/resource_test_result1.png)
+
+the 12.99% in the figure above means that 12.99% of one CPU core was consumed.
+
+**Results and Conclusion: When APP A fully utilizes one CPU core, the eBPF program consumes 1.73%(totally 13.9% usage of one CPU core  == 1.73% of 8 CPU cores) of the CPU, which is less than the POD CPU limit of 12.5%. There are two possible reasons.**
+
+- APP A and Kmesh eBPF share the POD CPU limit, with Kmesh eBPF CPU being restricted by the POD CPU limit.
+- It is possible that due to the high performance of eBPF, APP A is not generating enough load to cause eBPF to exceed the CPU limit, further experiments in section 3.2 are needed.
```

Comment:
You can say this indicates ebpf cpu cost is fairly small compared to application itself.

I am also unsure if there are other applications in the same node sharing the bpf prog